File: CSharp/Builder/Taco.cs
namespace Builder
{
    public class Taco
    {
        public string? Meat { get; set; } = "None";
        public string? Cheese { get; set; } = "None";
        public string? Shell { get; set; } = "None";
        public bool? Vegan { get; set; } = false;
        public bool? Salad { get; set; } = false;



        public class Builder
        {
            private readonly Taco _taco = new();

            public Builder Meat(string meat)
            {
                _taco.Meat = meat;
                return this;
            }

            public Builder Cheese(string cheese)
            {
                _taco.Cheese = cheese;
                return this;
            }

            public Builder Shell(string shell)
            {
                _taco.Shell = shell;
                return this;
            }

            public Builder Vegan(bool vegan)
            {
                _taco.Vegan = vegan;
                return this;
            }

            public Builder Salad(bool salad)
            {
                _taco.Salad = salad;
                return this;
            }

            public Taco Build()
            {
                return _taco;
            }
        }

        public override string ToString()
        {
            return $"Taco\n{nameof(Meat)}: {Meat}, {nameof(Cheese)}: {Cheese}, {nameof(Shell)}: {Shell}, {nameof(Vegan)}: {Vegan}, {nameof(Salad)}: {Salad}";
        }
    }
}

File: Node.js/Taco.js
class Taco {
  #meat = null;
  #cheese = null;
  #shell = null;
  #vegan = false;
  #salad = false;

  static Builder = class {
    #meat = null;
    #cheese = null;
    #shell = null;
    #vegan = false;
    #salad = false;

    meat(meat) {
      this.#meat = meat;
      return this;
    }

    cheese(cheese) {
      this.#cheese = cheese;
      return this;
    }

    shell(shell) {
      this.#shell = shell;
      return this;
    }

    vegan(vegan) {
      this.#vegan = vegan;
      return this;
    }

    salad(salad) {
      this.#salad = salad;
      return this;
    }

    build() {
      const taco = new Taco(
        this.#meat,
        this.#cheese,
        this.#shell,
        this.#vegan,
        this.#salad
      );
      return taco;
    }
  };

  constructor(meat, cheese, shell, vegan, salad) {
    this.#meat = meat;
    this.#cheese = cheese;
    this.#shell = shell;
    this.#vegan = vegan;
    this.#salad = salad;
  }

  toString() {
    return `
            Meat: ${this.#meat}
            Cheese: ${this.#cheese}
            Shell: ${this.#shell}
            Vegan: ${this.#vegan}
            Salad: ${this.#salad}
        `;
  }
}

export default Taco;
